repo_validation/README.md

```diff
@@ -0,0 +1,338 @@
+cmake -S . -B build -Drun_repo_validation=ON -Dfix_repo_validation_errors=ON
+```
+
+### Running Validation
+
+To run all validation checks:
+
+```bash
+# Using CMake (validation only, report errors)
+cmake --build . --target your_project_name_repo_validation
+
+# Or using Visual Studio
+# Build the "your_project_name_repo_validation" target from the solution
+```
+
+### Auto-Fixing Validation Errors
+
+When you configure CMake with `-Dfix_repo_validation_errors=ON`, the validation scripts will automatically attempt to fix any validation errors they encounter:
+
+```bash
+# Configure with fix mode enabled
+cmake -S . -B build -Drun_repo_validation=ON -Dfix_repo_validation_errors=ON
+
+# Run validation (will fix errors)
+cmake --build build --target your_project_name_repo_validation
+```
+
+**Note**: The `-Fix` parameter is automatically passed to all validation scripts when `fix_repo_validation_errors=ON`.
+
+## Available Validations
+
+### File Ending Newline Validation
+
+**Script:** `scripts/validate_file_endings.ps1`
+
+**Purpose:** Ensures all source code files (`.h`, `.hpp`, `.c`, `.cpp`, `.cs`) end with a newline character (CRLF on Windows).
+
+**Exclusions:** 
+- Default exclusions (if not customized): `deps`, `cmake`
+- Custom exclusions can be specified via `EXCLUDE_FOLDERS` parameter in `add_repo_validation()`
+
+**Rationale:** Source files must end with a newline to comply with:
+- C/C++ language standards (files should end with a newline)
+- Compiler warnings and requirements
+- Git and version control best practices
+- Consistent file formatting across the codebase
+
+**Fix Mode:** When run with `-Fix` parameter, the script automatically corrects files:
+- Adds CRLF to files that end abruptly without any newline
+- Converts LF-only endings to CRLF (Windows standard)
+- Converts CR-only endings to CRLF
+- **Does not modify any files in excluded directories**
+
+**Manual Fix Options:**
+- In Visual Studio: Ensure the cursor can move one line past the last line of code
+- In VS Code: Add a final newline at the end of the file
+- Configure `.editorconfig` with `insert_final_newline = true`
+
+### Requirements Document Naming Validation
+
+**Script:** `scripts/validate_requirements_naming.ps1`
+
+**Purpose:** Ensures that requirement documents in `devdoc` folders follow the naming convention `{module_name}_requirements.md`.
+
+**Detection:** A markdown file in a `devdoc` folder is considered a requirements document if it contains SRS (Software Requirements Specification) tags matching the pattern: `SRS_{MODULE}_{DEVID}_{REQID}` (e.g., `SRS_MY_MODULE_01_001`)
+
+**Exclusions:**
+- Default exclusions (if not customized): `deps`, `cmake`
+- Custom exclusions can be specified via `EXCLUDE_FOLDERS` parameter in `add_repo_validation()`
+
+**Rationale:** Consistent naming conventions for requirement documents:
+- Makes requirements easier to locate by module name
+- Clearly identifies files as containing formal requirements
```

Comment:
as an aside, I sometimes wonder if it would have been better to have like a prefix to differentiate "requirements" and "design" files. Sometimes it's hard to filter through the noise looking for a design doc in the "devdocs" folder.